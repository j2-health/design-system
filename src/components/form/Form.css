.ant-form-item .ant-form-item-label > label {
  align-items: start;
  padding-top: 8px;

  &::before {
    margin-top: 6px;
  }
}
