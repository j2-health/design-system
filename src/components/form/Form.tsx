import { FormProps } from 'antd'
import { Form as AntdForm, FormItem } from 'formik-antd'

type Props = Expand<FormProps>

export const Form = ({ children, ...props }: Props) => {
  return <AntdForm {...props}>{children}</AntdForm>
}

Form.Item = FormItem
