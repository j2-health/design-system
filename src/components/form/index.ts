export { Form } from 'formik-antd'
