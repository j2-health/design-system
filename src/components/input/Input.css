.j2-input-number {
  &.ant-input-number {
    .ant-input-number-handler-wrap {
      width: auto;

      .ant-input-number-handler {
        height: 50%;
        padding: 0 12px;
        color: var(--j2-color-primary);
      }

      .ant-input-number-handler-up {
        align-items: flex-end;
      }

      .ant-input-number-handler-down {
        border-block-start: none;
        align-items: flex-start;
      }
    }
  }
}
