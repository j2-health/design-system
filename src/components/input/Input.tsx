import { Input as FormikAntdInput, InputProps } from 'formik-antd'
type Props = Expand<InputProps>

export const Input = ({ size = 'large', ...props }: Props) => {
  return <FormikAntdInput size={size} {...props} />
}
