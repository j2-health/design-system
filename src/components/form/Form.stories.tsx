import type { Meta, StoryObj } from '@storybook/react'
import { Formik } from 'formik'
import { Form } from './Form'
import { Select } from '../select'
import { Button } from '../button/Button'
import { action } from '@storybook/addon-actions'
import * as Yup from 'yup'

const meta = {
  title: 'Components/Form',
  component: Form,
} satisfies Meta<typeof Form>

export default meta
type Story = StoryObj<typeof meta>

export const BasicForm: Story = {
  render: () => {
    return (
      <Formik
        initialValues={{ ma_plan_type: '' }}
        validationSchema={Yup.object({
          ma_plan_type: Yup.string().required('MA plan type is required'),
        })}
        onSubmit={(values) => action('form-submitted')(values)}
      >
        {({ isValid, dirty }) => {
          const isFormReadyToSubmit = isValid && dirty
          return (
            <Form>
              <Form.Item label="MA Plan Type" name="ma_plan_type" required>
                <Select
                  name="ma_plan_type"
                  placeholder="Select a plan type"
                  allowClear={true}
                  style={{ width: '256px' }}
                  options={[
                    { label: 'HMO', value: 'hmo' },
                    { label: 'HMO SNP', value: 'hmo_snp' },
                    { label: 'HMO-POS', value: 'hmo_pos' },
                    { label: 'HMO D-SNP', value: 'hmo_d_snp' },
                    { label: 'PPO', value: 'ppo' },
                    { label: 'PPO SNP', value: 'ppo_snp' },
                    { label: 'PFFS', value: 'pffs' },
                    { label: 'MSA', value: 'msa' },
                    { label: 'Cost Plan', value: 'cost_plan' },
                    { label: 'PACE', value: 'pace' },
                  ]}
                />
              </Form.Item>
              <Button
                type="primary"
                htmlType="submit"
                disabled={!isFormReadyToSubmit}
              >
                Submit
              </Button>
            </Form>
          )
        }}
      </Formik>
    )
  },
}
