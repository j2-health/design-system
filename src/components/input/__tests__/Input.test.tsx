import { render } from '@testing-library/react'
import { Input } from '../Input'
import { Formik } from 'formik'
import { Form } from '../../form'

describe('Input', () => {
  const renderInForm = (component: React.ReactNode) => {
    return render(
      <Formik initialValues={{ name: '' }} onSubmit={() => {}}>
        <Form>{component}</Form>
      </Formik>
    )
  }

  it('should render correctly', () => {
    const { container } = renderInForm(<Input name="name" />)
    expect(container).toMatchSnapshot()
  })
})
