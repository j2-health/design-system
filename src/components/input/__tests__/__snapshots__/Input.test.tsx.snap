// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal css-dev-only-do-not-override-os2fvf"
  >
    <input
      class="ant-input ant-input-lg css-dev-only-do-not-override-os2fvf ant-input-outlined"
      id="name"
      name="name"
      type="text"
      value=""
    />
  </form>
</div>
`;
